designcompose/src/main/java/com/android/designcompose/DocServer.kt

```diff
@@ -447,20 +450,24 @@ internal fun DocServer.doc(
                 }
                 targetDoc
             }
+        docUpdateCallback?.invoke(targetDoc?.c?.toSerializedBytes(Feedback))
```

Comment:
Regarding the discussion around registering callbacks:

@timothyfroehlich, based on this change, the serialization won't happen if docUpdateCallback is null.

i.e. the code doesn't unnecessarily generate that data.

Reply:
Agreed. What I'm saying is that having a "docUpdateCallback" is a generically useful feature, but it should not automatically serialize the doc whenever it is called. I'm sure there are use cases where an app developer would want to react to the doc updating without wanting to do anything with the serialized file.

Wouldn't it be much more useful to add a function to provide the serialized file on demand, rather than requiring the user to register a callback? I'm asking for a better generic design, not one that fulfills only exactly Umesh's use case.

Reply:
If LiveUpdate is disabled, is this where the document would be loaded from disk and the callback invoked?

Reply:
Yes